data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/JacksonEvent.java

```diff
@@ -232,6 +245,25 @@ public EventMetadata getMetadata() {
         return eventMetadata;
     }
 
+    @Override
+    public boolean containsKey(final String key) {
+
+        final String trimmedKey = checkAndTrimKey(key);
+
+        final JsonNode node = getNode(trimmedKey);
+
+        return !node.isMissingNode();
+    }
+
+    @Override
+    public boolean isValueAList(final String key) {
+        try {
+            return getList(key, Object.class) != null;
```

Comment:
QUES: Since this method just checks the list type, is it possible that we can just retrieve the JsonNode value and utilize JsonNode::isArray() without actually converting it to a list in this method?

Reply:
I like the suggestion.